### PR TITLE
Add support for `snow://` stage paths

### LINKED ISSUE #1345: SNOW-1282839: Add support for SnowURLs
## What is the current behavior?

Unable to use `snow://` URLs with Snowpark Python APIs like `Session.file.get_stream()`

## What is the desired behavior?

Able to access files with `snow://` URLs.

## How would this improve `snowflake-snowpark-python`?

Compatibility with new Snowflake stage URL type.

### PR DESCRIPTION
**Pull request by sfc-gh-dhung**
Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   Fixes #1345

2. Fill out the following pre-review checklist:

   - [x] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

Adds support for `snow://` prefixed stage paths. Also fixes a bug where local file path is incorrectly extracted from single quoted paths: `'@path/file'` -> `file'` (note trailing quote)

### CODE REVIEW

File: src/snowflake/snowpark/file_operation.py [hunk 1]
```diff
@@ -338,7 +339,9 @@ def get_stream(
         else:
             options = {"parallel": parallel}
             tmp_dir = tempfile.gettempdir()
-            src_file_name = stage_location.rsplit("/", maxsplit=1)[1]
+            src_file_name = unwrap_single_quote(stage_location).rsplit("/", maxsplit=1)[
+                1
+            ]
```
Comment (sfc-gh-aalam) [inline src/snowflake/snowpark/file_operation.py:344]: does this change also needs to apply to `get` / `put` / `put_stream`? 

note to self: run stored-proc test before merging

**Comment by sfc-gh-dhung** [comment]
I have read the CLA Document and I hereby sign the CLA

File: src/snowflake/snowpark/file_operation.py (hunk 1 above)
Comment (sfc-gh-dhung) [reply src/snowflake/snowpark/file_operation.py:344]: `get()` seems to work fine with just the `utils` change. Still working on testing the sproc path, will also look into the `put`/`put_stream` paths.

Comment (sfc-gh-dhung) [reply src/snowflake/snowpark/file_operation.py:344]: one note: many `snow://` stage types don't allow write/`PUT`. It's still worth ensuring the client code works but that path may not work end to end

Comment (sfc-gh-dhung) [reply src/snowflake/snowpark/file_operation.py:346]: Applied relevant changes to `put_stream`.

**Review by sfc-gh-aalam — APPROVED** [review]
LGTM. Running SP tests here: https://ci-dev-142.int.snowflakecomputing.com/job/PythonStoredProcBuildPrecommitTest/1001/

**Comment by sfc-gh-aalam** [comment]
> LGTM. Running SP tests here: https://ci-dev-142.int.snowflakecomputing.com/job/PythonStoredProcBuildPrecommitTest/1001/

Relevant tests related to this change passed. Only failures are also open issues right now
- t_python_sp.enable_fix_1022487.ssql
- t_python_sp.wametrics_tracking_py_sp.py

**Review by sfc-gh-stan — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -37,10 +37,12 @@
     - show
   - snowflake.snowpark.DataFrameWriter:
     - save_as_table
+- Added support for snow:// URLs to `snowflake.snowpark.Session.file.get` and `snowflake.snowpark.Session.file.get_stream`
```
Comment (sfc-gh-stan) [inline CHANGELOG.md:40]: Is there support for PUT too?

Comment (sfc-gh-dhung) [reply CHANGELOG.md:40]: Technically snow:// URLs will be correctly parsed for PUT but writing to snow:// stages is limited

**Pull request merged**